src/hotspot/share/opto/mulnode.cpp

```diff
@@ -670,9 +670,11 @@ const Type *AndINode::mul_ring( const Type *t0, const Type *t1 ) const {
   return and_value<TypeInt>(r0, r1);
 }
 
+static bool AndIL_is_zero_element_under_mask(const PhaseGVN* phase, const Node* expr, const Node* mask, BasicType bt);
```

Comment:
Why do you split declaration and definition? Could the body just be moved up here?

Reply:
Because of the details wrt ADD the body feels out of context here. I find it easier to read when it's close to  `AndIL_sum_and_mask` below.